packages/@react-aria/dnd/src/DragManager.ts

```diff
@@ -38,7 +38,7 @@ interface DropTarget {
 export function registerDropTarget(target: DropTarget) {
   dropTargets.set(target.element, target);
   dragSession?.updateValidDropTargets();
-  return () => {
+  return (): void => {
```

Comment:
oh this will be really annoying

Reply:
yes, it can go on either because --isolatedDeclarations will work if the return value is trivial to determine. I do not know exactly what constitutes trivial. But you can search this page for it https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-5.html#isolated-declarations

Likely the eslint rule is catching too many because it doesn't know what is trivial or not. The hard part of doing this as typescript first though is that it's all or nothing whereas I was able to focus on changes with the eslint error.